pwndbg/gdblib/symbol.py

```diff
@@ -243,14 +243,14 @@ def address(symbol: str) -> int:
 
 @pwndbg.lib.memoize.reset_on_objfile
 @pwndbg.lib.memoize.reset_on_thread
-def static_linkage_symbol_address(symbol):
-    if isinstance(symbol, int):
-        return symbol
+def static_linkage_symbol_address(symbol: str) -> int:
+    """
+    Get the address for static linkage `symbol`
+    """
 
-    try:
-        return int(symbol, 0)
-    except Exception:
-        pass
+    # GDB < 9.x does not have `gdb.lookup_static_symbol`
+    if not hasattr(gdb, "lookup_static_symbol"):
+        return None
```

Comment:
Can't we fallback to `gdb.lookup_symbol` in such case? or does this happen elsewhere?

Ideally, we do not want to make this check here and instead e.g. create a function based on GDB version like:
```py
def _get_symbol_address_gdb_9x(...): ...
def _get_symbol_address_gdb_10x(...): ...

static_linkage_symbol_address = _get_symbol_address_gdb_10x if gdb_version >= 10 else _get_symbol_address_gdb_9x
```

Reply:
> Can't we fallback to `gdb.lookup_symbol` in such case?

Hmm, yes and no, because `gdb.lookup_symbol` might not have the same results as `gdb.lookup_static_symbol`.

> Ideally, we do not want to make this check here and instead e.g. create a function based on GDB version like:

Yea, I also think what you said will be better, but I didn't find a way to make GDB < 9.x have the same results as `gdb.lookup_static_symbol`

Reply:
Iirc the drawback of `gdb.lookup_symbol` is that we may find incorrect symbol, defined as a type in the program code, is that right? Though that won't happen that often.

If this is what it was about, then we should still fallback to `gdb.lookup_symbol` because with the current code we will not find the symbol at all, since we return `None`? Or am I missing something here and this would somehow magically work in the end?

Reply:
> Iirc the drawback of `gdb.lookup_symbol` is that we may find incorrect symbol, defined as a type in the program code, is that right? Though that won't happen that often.

Yes, you are right, this is the only drawback of it.

> If this is what it was about, then we should still fallback to `gdb.lookup_symbol` because with the current code we will not find the symbol at all, since we return `None`?

Yeah, it seems better that we fallback to `gdb.lookup_symbol` instead of just returning `None` here, although there's a little drawback. I'll make it fallback to `gdb.lookup_symbol` until we find a better solution.